lib/client/src/client.rs

```diff
@@ -1809,12 +1809,13 @@ impl TransactionBuilder {
         rpc: &RpcClientAsync,
     ) -> anyhow::Result<solana_sdk::transaction::VersionedTransaction> {
         let latest_blockhash = rpc.get_latest_blockhash().await?;
-        self.transaction_with_blockhash(latest_blockhash)
+        self.transaction_with_blockhash(latest_blockhash, Option::None)
     }
 
     pub fn transaction_with_blockhash(
         &self,
         blockhash: Hash,
+        compute_unit_limit_opt: Option<u32>,
```

Comment:
instead of adding it as an optional argument, how about adding a `prepend_compute_limit_instruction()` function to TransactionBuilder?

Reply:
how about doing this instead

![image](https://github.com/blockworks-foundation/mango-v4/assets/89031858/ea2ff511-01c6-4cd7-9ea2-1ae56c785be5)

Reply:
I'd like to keep client specific logic out of `TransactionBuilder`. How about this:
- Add `default_compute_budget_per_instruction` field to `TransactionBuilderConfig`. The client sets this to Some(250k) by default
- In transaction_with_blockhash: check if a compute budget / compute unit price ix exists, if not, add it with the values from TransactionBuilderConfig.

Reply:
pushed fixup